src/openforms/js/components/admin/form_design/MissingComponentTranslationsWarning.js

```diff
@@ -0,0 +1,141 @@
+import FormioUtils from 'formiojs/utils';
+import PropTypes from 'prop-types';
+import React, {useState} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import MessageList from 'components/admin/MessageList';
+import Modal from 'components/admin/Modal';
+import {ChangelistColumn, ChangelistTable} from 'components/admin/tables';
+import {
+  extractComponentLiterals,
+  getSupportedLanguages,
+} from 'components/formio_builder/translation';
+
+const LANGUAGES = getSupportedLanguages();
+
+const extractTranslatableValues = configuration => {
+  let translatableValues = [];
+  FormioUtils.eachComponent(
+    configuration.components,
+    component => {
+      const literals = extractComponentLiterals(component);
+      literals.forEach(literal => {
+        translatableValues.push({
+          componentKey: component.key,
+          componentLabel: component.label,
+          literal: literal,
+        });
+      });
+    },
+    true
+  );
+  return translatableValues;
+};
+
+const MissingComponentTranslationsTable = ({children: missingTranslations}) => (
+  <ChangelistTable data={missingTranslations}>
+    <ChangelistColumn objProp="componentKey">
+      <FormattedMessage description="Key of the Form component" defaultMessage="Component key" />
+    </ChangelistColumn>
+
+    <ChangelistColumn objProp="componentLabel">
+      <FormattedMessage
+        description="Label of the Form component"
+        defaultMessage="Component label"
+      />
+    </ChangelistColumn>
+
+    <ChangelistColumn objProp="language">
+      <FormattedMessage description="Readable label for the language" defaultMessage="Language" />
+    </ChangelistColumn>
+
+    <ChangelistColumn objProp="literal">
+      <FormattedMessage description="Literal to be translated" defaultMessage="Literal" />
+    </ChangelistColumn>
+  </ChangelistTable>
+);
+
+MissingComponentTranslationsTable.propTypes = {
+  children: PropTypes.arrayOf(
+    PropTypes.shape({
+      componentKey: PropTypes.string.isRequired,
+      componentLabel: PropTypes.string.isRequired,
+      language: PropTypes.string.isRequired,
+      literal: PropTypes.string.isRequired,
+    })
```

Comment:
PropTypes.object is only good enough when you don't know the shape of the object. Whenever we do (as evidenced here by the prop names that we're reading out in the component), make this explicit.